lib/Runtime/Language/DynamicProfileInfo.cpp

```diff
@@ -506,16 +506,23 @@ namespace Js
             funcBody->SetCallbackInfoList(list);
         }
 
-        CallbackInfo * info = FindCallbackInfo(funcBody, callSiteId);
-        if (info == nullptr)
+        CallbackInfoList::EditingIterator iter = list->GetEditingIterator();
+        while (iter.Next())
         {
-            info = RecyclerNewStructZ(funcBody->GetScriptContext()->GetRecycler(), CallbackInfo);
-            info->callSiteId = callSiteId;
-            info->sourceId = NoSourceId;
-            info->canInlineCallback = true;
-            list->Prepend(info);
+            Field(CallbackInfo*) callbackInfo = iter.Data();
+            if (callbackInfo->callSiteId == callSiteId)
+            {
+                return callbackInfo;
+            }
         }
```

Comment:
nit: you could also have FindCallBackInfo return an iterator. If the iterator is valid (Iterator::IsValid), you're done, otherwise add a new CallBackInfo.

Reply:
FindCallBackInfo has to be able to return nullptr if the CallbackInfoList is null, but a common helper can return an iterator, yeah.